platform/gce.go

```diff
@@ -44,10 +46,11 @@ type GCEOptions struct {
 }
 
 type gceCluster struct {
-	api      *compute.Service
-	sshAgent *network.SSHAgent
-	conf     *GCEOptions
-	machines map[string]*gceMachine
+	api        *compute.Service
+	sshAgent   *network.SSHAgent
+	conf       *GCEOptions
+	machines   map[string]*gceMachine
+	sync.Mutex // protects concurrent access to machines
```

Comment:
can you name this 'mu' instead of embedding it? i'd like to move each platform's code to its own package, and when that happens, gceCluster will need to be exported as GCECluster. let's prevent accidents by not exposing Lock/Unlock there.